src/oc_reporter.erl

```diff
@@ -123,7 +128,14 @@ send_spans(Reporter, Config) ->
             ok;
         Spans ->
             ets:delete_all_objects(Buffer),
-            report(Reporter, Spans, Config)
+            %% don't let a reporter exception crash us
+            try
+                report(Reporter, Spans, Config)
+            catch
+                _Class:_Exception ->
+                    error_logger:info_msg("reporter threw exception: reporter=~p stacktrace=~p",
```

Comment:
The error logger could be risky here if this happens a lot we might just DOS the system with it.

Reply:
True. I really don't know what to do with logs in Erlang until OTP improves :(. Should we just not log it, accept the risky or add a backoff?

Reply:
I honestly am not sure, I just noticed it and thought it'd make sense to bring it up, while I really like having the option of knowing errors but it worries me that it could bring down the system. Adding lager feels like quite a heavy-handed solution. Would add a simple gen server that handles logs and just sheds them when there is something in a queue an option?

Reply:
Hm, with lager's error_logger handler would this still DOS the system? I'm guessing so since it still goes through the event handler? Wanted to make sure.

I think adding a gen server for log shedding is as much overhead as just adding lager ;).

Reply:
true, I think the lager loggers protect against overload. you'll get "x messags shed" if you're producing them too rapidly. not sure about the error logger handler perse so